big_o/test/test_complexityclass_comparisons.py

```diff
@@ -0,0 +1,44 @@
+import unittest
+import numpy as np
+from numpy.testing import assert_array_almost_equal
+
+from big_o.complexities import ComplexityClass
+
+
+class FirstComplexityClass(ComplexityClass):
+    order = 1
+
+
+class SecondComplexityClass(ComplexityClass):
+    order = 2
+
+
+FirstComplexity = FirstComplexityClass()
+SecondComplexity = SecondComplexityClass()
+
+
+class TestComplexities(unittest.TestCase):
+
+    def test_ge(self):
+        self.assertEqual(FirstComplexity >= SecondComplexity, False)
+        self.assertEqual(FirstComplexity >= FirstComplexity, True)
+        self.assertEqual(SecondComplexity >= FirstComplexity, True)
+
+    def test_le(self):
+        self.assertEqual(FirstComplexity <= SecondComplexity, True)
+        self.assertEqual(FirstComplexity <= FirstComplexity, True)
+        self.assertEqual(SecondComplexity <= FirstComplexity, False)
+
+    def test_l(self):
+        self.assertEqual(FirstComplexity < SecondComplexity, True)
+        self.assertEqual(FirstComplexity < FirstComplexity, False)
+        self.assertEqual(SecondComplexity < FirstComplexity, False)
+
+    def test_g(self):
+        self.assertEqual(FirstComplexity > SecondComplexity, False)
+        self.assertEqual(FirstComplexity > FirstComplexity, False)
+        self.assertEqual(SecondComplexity > FirstComplexity, True)
+
+    def test_eq(self):
+        self.assertEqual(FirstComplexity == SecondComplexity, False)
+        self.assertEqual(FirstComplexity == FirstComplexity, True)
```

Comment:
To be clear again about the `<=`, `>=`, and `==`, I'm attaching a modified version of the tests to show how I expect the classes to behave. As you say, we can discuss whether the `==` test should pass, but I think the `>=` and `<=` tests should definitively pass (they are failing now)

Reply:
```
import unittest

from big_o.complexities import ComplexityClass


class FirstComplexityClass(ComplexityClass):
    order = 1


class AltFirstComplexityClass(ComplexityClass):
    order = 1


class SecondComplexityClass(ComplexityClass):
    order = 2


class TestComplexities(unittest.TestCase):

    def setUp(self):
        self.first_complexity = FirstComplexityClass()
        self.alt_first_complexity = AltFirstComplexityClass()
        self.second_complexity = SecondComplexityClass()

    def test_ge(self):
        self.assertFalse(self.first_complexity >= self.second_complexity)
        self.assertTrue(self.second_complexity >= self.first_complexity)
        self.assertTrue(self.first_complexity >= self.first_complexity)
        self.assertTrue(self.alt_first_complexity >= self.first_complexity)
        self.assertTrue(self.first_complexity >= self.alt_first_complexity)

    def test_le(self):
        self.assertTrue(self.first_complexity <= self.second_complexity)
        self.assertFalse(self.second_complexity <= self.first_complexity)
        self.assertTrue(self.first_complexity <= self.first_complexity)
        self.assertTrue(self.alt_first_complexity <= self.first_complexity)
        self.assertTrue(self.first_complexity <= self.alt_first_complexity)

    def test_l(self):
        self.assertTrue(self.first_complexity < self.second_complexity)
        self.assertFalse(self.second_complexity < self.first_complexity)
        self.assertFalse(self.first_complexity < self.alt_first_complexity)

    def test_g(self):
        self.assertFalse(self.first_complexity > self.second_complexity)
        self.assertTrue(self.second_complexity > self.first_complexity)
        self.assertFalse(self.first_complexity > self.alt_first_complexity)

    def test_eq(self):
        self.assertFalse(self.first_complexity == self.second_complexity)
        self.assertTrue(self.first_complexity == self.first_complexity)
        self.assertTrue(self.first_complexity == self.alt_first_complexity)
```

Reply:
Tests now pass.